### PR TITLE
Update dockerfiles to enable LLVM exception handling variable

### PR DESCRIPTION
**Pull request by prigoyal**
first simply updating the docs and dockerfile to have correct llvm image, and then will change the halide build in another PR

### CODE REVIEW

File: docker/linux-trusty-gcc4.8-tapir5.0/Dockerfile [hunk 1]
```diff
@@ -19,12 +19,12 @@ ENV CXX /usr/bin/g++
...
 WORKDIR $LLVM_SOURCES
 
-ENV CLANG_PREFIX /usr/local/clang+llvm-tapir5.0
+ENV CLANG_PREFIX $HOME/clang+llvm-tapir5.0
 ENV CMAKE_VERSION cmake
 
 RUN git clone --recursive https://github.com/wsmoses/Tapir-LLVM llvm && \
     mkdir -p ${LLVM_SOURCES}/llvm_build && cd ${LLVM_SOURCES}/llvm_build && \
-    ${CMAKE_VERSION} -DCMAKE_INSTALL_PREFIX=${CLANG_PREFIX} -DLLVM_TARGETS_TO_BUILD=X86 -DCOMPILER_RT_BUILD_CILKTOOLS=OFF -DLLVM_ENABLE_CXX1Y=ON -DLLVM_ENABLE_TERMINFO=OFF -DLLVM_BUILD_TESTS=OFF -DLLVM_ENABLE_ASSERTIONS=ON -DCMAKE_BUILD_TYPE=Release -DLLVM_BUILD_LLVM_DYLIB=ON  -DLLVM_ENABLE_RTTI=ON ../llvm/ && \
+    ${CMAKE_VERSION} -DLLVM_REQUIRES_EH=ON -DLLVM_ENABLE_OCAMLDOC=OFF -DLLVM_INSTALL_OCAMLDOC_HTML_DIR=/tmp -DLLVM_OCAML_INSTALL_PATH=/tmp -DCMAKE_INSTALL_PREFIX=${CLANG_PREFIX} -DLLVM_TARGETS_TO_BUILD=X86 -DCOMPILER_RT_BUILD_CILKTOOLS=OFF -DLLVM_ENABLE_CXX1Y=ON -DLLVM_ENABLE_TERMINFO=OFF -DLLVM_BUILD_TESTS=OFF -DLLVM_ENABLE_ASSERTIONS=ON -DCMAKE_BUILD_TYPE=Release -DLLVM_BUILD_LLVM_DYLIB=ON  -DLLVM_ENABLE_RTTI=ON ../llvm/ && \
```
Comment (abadams) [inline docker/linux-trusty-gcc4.8-tapir5.0/Dockerfile]: I thought the flag was LLVM_ENABLE_EH. Are you sure it's REQUIRES_EH?

Comment (prigoyal) [reply docker/linux-trusty-gcc4.8-tapir5.0/Dockerfile]: both are supported
https://github.com/wsmoses/Tapir-LLVM/blob/6d439c5b676577f48e20d54edef477d1a6c517bc/cmake/modules/AddLLVM.cmake#L13

Comment (prigoyal) [reply docker/linux-trusty-gcc4.8-tapir5.0/Dockerfile]: but I am not sure why there exists two flags though, and it feels like REQUIRES_EH might be enabling something more?

**Comment by abadams** [comment]
That file claims that REQUIRES is an internal thing for other LLVM build
targets to use when they require exceptions. I think publicly-facing option
is ENABLE

**Comment by prigoyal** [comment]
alright, I'll change to ENABLE now

**Comment by prigoyal** [comment]
done!

**Comment by prigoyal** [comment]
can I get the stamp on this now @abadams ? :)

**Pull request merged**